specification/cognitiveservices/data-plane/Language/preview/2021-11-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,531 @@
+      "required": [
+        "query",
+        "prediction"
+      ],
+      "properties": {
+        "query": {
+          "description": "The same query given in request.",
+          "type": "string"
+        },
+        "detectedLanguage": {
+          "description": "The detected language from the query.",
+          "type": "string"
+        },
+        "prediction": {
+          "description": "The predicted result for the query.",
+          "$ref": "#/definitions/ConversationPrediction"
+        }
+      }
+    },
+    "ConversationPrediction": {
+      "type": "object",
+      "description": "Represents the prediction section of a Conversation project.",
+      "x-ms-discriminator-value": "conversation",
+      "required": [
+        "intents",
+        "entities"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/BasePrediction"
+        }
+      ],
+      "properties": {
+        "intents": {
+          "description": "The intent classification results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationIntent"
+          }
+        },
+        "entities": {
+          "description": "The entity extraction results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationEntity"
+          }
+        }
+      }
+    },
+    "ConversationIntent": {
+      "type": "object",
+      "description": "The intent classification result of a Conversation project.",
+      "required": [
+        "category",
+        "confidenceScore"
+      ],
+      "properties": {
+        "category": {
+          "description": "A predicted class.",
+          "type": "string"
+        },
+        "confidenceScore": {
+          "format": "float",
+          "description": "The confidence score of the class from 0.0 to 1.0.",
+          "type": "number",
+          "minimum": 0,
+          "maximum": 1
+        }
+      }
+    },
+    "ConversationEntity": {
+      "type": "object",
+      "description": "The entity extraction result of a Conversation project.",
+      "required": [
+        "category",
+        "text",
+        "offset",
+        "length",
+        "confidenceScore"
+      ],
+      "properties": {
+        "category": {
+          "description": "The entity category.",
+          "type": "string"
+        },
+        "text": {
+          "description": "The predicted entity text.",
+          "type": "string"
+        },
+        "offset": {
+          "format": "int32",
+          "description": "The starting index of this entity in the query.",
+          "type": "integer"
+        },
+        "length": {
+          "format": "int32",
+          "description": "The length of the text.",
+          "type": "integer"
+        },
+        "confidenceScore": {
+          "format": "float",
+          "description": "The entity confidence score.",
+          "type": "number"
+        }
+      },
+      "additionalProperties": true
```

Comment:
This is newly added because we want to include extra information here. For example, we will add a `listKeys` in the public preview version. Is this the preferred way to include it?

```
{
  "category": "artist",
  "text": "Joe",
  "offset": 17,
  "length": 7,
  "confidenceScore": 0.99691725,
  "listKeys": ["Joe D."]
}
```

Reply:
So if I'm understanding this correctly: by setting `additionalProperties` to true, it enables the service to return extra properties that are not included in the swagger itself; like your example of including `listKeys` as an additional attribute for this entity.

I am concerned that this would cause issues. Firstly, it seems unintuitive for the property names to not be included in the swagger at all (`listKeys` in this example). Customers (and us) would need to refer to examples to figure out the name of the potential properties (unless I'm missing something). Secondly, I'm not sure how Autorest (the code generation tool we use for SDKs) would handle this since it relies on information within the swagger itself. @heaths might be able to shed some light on this.

Would it be better to include `listKeys` as an optional attribute for a `ConversationEntity` within the swagger?